Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy.swift

```diff
@@ -426,6 +427,10 @@ extension PathHierarchy {
         fileprivate(set) unowned var parent: Node?
         /// The symbol, if a node has one.
         fileprivate(set) var symbol: SymbolGraph.Symbol?
+
+        /// The node representing this node's overload group, if it's part of one.
+        fileprivate(set) unowned var overloadGroup: Node?
```

Comment:
I don't feel like this is a good fit for the path hierarchy since it's modeling a special case hierarchy that:
- isn't represented in the symbol's path components
- isn't used for link resolution
- isn't reflected in the page's topic reference or web URL.

Because of these things, I wonder if this would be better represented in the topic graph.

Reply:
Without adding the information about a node's "effective parent", the path-hierarchy-based automatic curation wouldn't be able to curate individual overloaded symbols under their overload group. `PathHierarchyLinkResolver.directDescendants(of:languagesFilter:)` doesn't have access to the topic graph.

Reply:
I'm saying don't call `PathHierarchyLinkResolver` for overload groups since that curation isn't path based.